src/cpp/FileBuilder.cpp

```diff
@@ -0,0 +1,165 @@
+#include "FileBuilder.h"
+#include <cassert>
+
+
+ void FileBuilder::toFile(TextNote* note, std::string path) {
+	 std::ofstream file = createFile(note->title(), path);
+
+	 writeNote(file, note);
+	 file.close();
+
+}
+ std::ofstream FileBuilder::createFile(std::string name, std::string path) {
+	 std::ofstream file;
+	 file.open(getFileName(name, path));
+	 assert (!file.fail());
+	 return file;
+ }
+
+ void FileBuilder::writeNote(std::ofstream& file, TextNote* note) {
+	 writeCreationTime(file, note->_creation_time);
+	 writeTitle(file, note->title());
+	 writeText(file, note);
+ }
+
+
+ std::string FileBuilder::getFileName(std::string name, std::string path) {
+	 return path + name + "." + TEXT_FILE_EXTENSION;
+ }
+
+
+ void FileBuilder::writeTitle(std::ofstream& file, std::string title) {
+	 file << title << "\n";
+ }
+ void FileBuilder::writeCreationTime(std::ofstream& file, std::time_t creation_time) {
+ 	 file << toDateFormat(creation_time) << "\n";
+ }
+ void FileBuilder::writeText(std::ofstream& file, TextNote* note) {
+	 std::string line;
+	 std::stringstream s;
+
+	 s << note->text();
+
+	 while (std::getline(s, line)) {
+
+		 addEmptyLineCharIfEmpty(line);
+		 file << line << "\n";
+	 }
+
+	 file << "\n";
+ }
+void FileBuilder::addEmptyLineCharIfEmpty(std::string& line) {
+	 int i = 0;
+	 while(line[i] == EMPTY_LINE_CHAR) {
+		i++;
+	 }
+	 if (line[i] == '\0') {
+		 line += EMPTY_LINE_CHAR;
+	 }
+}
+TextNote FileBuilder::fromFile(std::string filename){
+	 std::ifstream file = openFile(filename);
+	 TextNote note = initTextNote(file);
+	 file.close();
+	 return note;
+ }
+
+std::ifstream FileBuilder::openFile(std::string filename) {
```

Comment:
This method is inconsistent API - the parameter is named filename, but it is actually collection title; there is no way to pass directory path, so it only supports collections in default location (path = ""). Changed in 1489ba108e513d413c88560b8344771f91cc56ed - now it accepts filename (which makes sense, we don't know title until we have opened file). Previous behavior can be achieved by explicitly calling getFileName (and passing both parameters if needed).